spec/spec_helper.rb

```diff
@@ -13,6 +15,16 @@
 rescue LoadError
 end
 
+def yaml_loader(path, spec_name=nil)
+  file =  File.expand_path(path)
+  raise "File #{file} cannot be found!" unless File.exist?(file)
+  raw_config = ERB.new(File.read(file)).result
+  raise "No configuration text found!" if raw_config.empty?
+  data = YAML.load(raw_config)
+  raise "No configuration data found!" if data.empty?
+  spec_name ? data[spec_name] : data
+end
```

Comment:
Well, this _could_ have been a one-liner:

    YAML.load(ERB.new(File.read(config_file_path)).result)

The problem is that there are three different kinds of errors that can occur:
* The `File.read` can fail
* The `ERB.new().result` can fail
* The `YAML.load()` can fail

And I wanted to make the specific failures easily identifiable.